src/runtime/components/Form.vue

```diff
@@ -155,9 +154,9 @@ provide('form-errors', errors)
 const inputs = ref<{ [P in keyof I]?: { id?: string, pattern?: RegExp } }>({})
 provide(formInputsInjectionKey, inputs as any)
 
-const dirtyFields = new Set<keyof I>()
-const touchedFields = new Set<keyof I>()
-const blurredFields = new Set<keyof I>()
+const dirtyFields: Ref<Set<keyof I>> = ref(new Set<keyof I>())
+const touchedFields: Ref<Set<keyof I>> = ref(new Set<keyof I>())
+const blurredFields: Ref<Set<keyof I>> = ref(new Set<keyof I>())
```

Comment:
We could use `reactive` instead of `ref` here

Reply:
Done, I had tried `reactive` before but was doing the types wrong.